File: server/src/main/java/com/nhom_5/server/entity/Combo.java
package com.nhom_5.server.entity;

import com.nhom_5.server.entity.enums.ComboStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "combo")
public class Combo extends BaseEntity {

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "price", nullable = false, precision = 12, scale = 2)
    private BigDecimal price;

    @Column(name = "image", columnDefinition = "TEXT")
    private String image;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    @Column(name = "status", nullable = false)
    private ComboStatus status = ComboStatus.ACTIVE;
}

File: client/src/types/combo.ts
export interface Combo {
  id: string
  name: string
  description: string
  price: number
  image: string | null
  status: 'ACTIVE' | 'INACTIVE'
}

export interface ComboPayload {
  name: string
  description?: string
  price: number
  image?: string
  status: 'ACTIVE' | 'INACTIVE'
}

export interface ComboSelection extends Combo {
  quantity: number
}
